lib/stellar_core_commander/process.rb

```diff
@@ -72,6 +103,22 @@ def close_ledger
       true
     end
 
+    Contract None => Hash
+    def metrics
+        response = server.get("/metrics") rescue false
+        if response
+          body = ActiveSupport::JSON.decode(response.body)
+          body["metrics"]
+        else
+          {}
+        end
+    end
+
+    Contract None => Num
+    def scp_ballots_prepared
+      metrics["scp.ballot.prepare"]["count"]
```

Comment:
This will raise an exception if `metrics` "fails" (i.e. the http request for metrics fails) and returns `{}`.  I would either remove the rescue clause from `metrics` and fail fast, or change this statement to be:

```ruby
sbp = metrics["scp.ballot.prepare"]
sbp.blank? ? 0 : sbp["count"]
```
or
```ruby
metrics["scp.ballot.prepare"].try(:[], "count")
```

`try` in the second example is like the Maybe monad... nil values will flow through.